src/Templates/ModelTemplate.cshtml

```diff
@@ -227,13 +227,13 @@ func (@(Model.Name.FixedValue.ToShortName()) *@(Model.Name)) UnmarshalJSON(body
     @EmptyLine
     v = m["@(p.SerializedName)"]
     if v != nil {
-            @if (p.ModelType is CompositeType && ((CompositeTypeGo) p.ModelType).IsPolymorphic)
+            @if (p.ModelType is CompositeTypeGo compositeType && compositeType.HasInterface)
```

Comment:
I see this pattern repeated in a few different places (and in other files).  It might be good to make this an extension on IModelType so you don't have to duplicate it.